File: pinyougou-manager-web/src/main/webapp/js/service/specificationService.js
//规格服务层
app.service('specificationService',function($http){
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../specification/search.do?page='+page+"&rows="+rows, searchEntity);
	} 
	
	//增加 
	this.add=function(entity){
		return  $http.post('../specification/add.do',entity );
	}
	
	//修改 
	this.update=function(entity){
		return  $http.post('../specification/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../specification/delete.do?ids='+ids);
	}
	
	//查询实体
	this.findOne=function(id){
		return $http.get('../specification/findOne.do?id='+id);
	}

	//规格下拉列表数据
	this.selectOptionList=function(){
		return $http.get('../specification/selectOptionList.do');
	}
});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.pojogroup.Specification;

import entity.PageResult;

/**
 * 服务层接口
 * 
 * @author Administrator
 *
 */
public interface SpecificationService {

	/**
	 * 返回全部列表
	 * 
	 * @return
	 */
	public List<TbSpecification> findAll();
	
	/**
	 * 分页+条件查询（条件封装在对象里）
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbSpecification specification, int pageNum,int pageSize);
	
	public void add(Specification specification);
	
	public Specification findOne(Long id);
	
	public void update(Specification specification);
	
	public void delete(Long[] ids);
	
	/**
	 * 返回规格下拉列表数据
	 * @return
	 */
	public List<Map> selectOptionList();

}
